data/terrarium.py

```diff
@@ -124,47 +125,41 @@ def getTrianglesFromPoints(P):
     return triangles
 
 # Given a set of points and height of the same lenght compose a voronoi PNG image
-def makeHeighmap(path, name, size, height_range, points, heights):
-    bbox = getBoundingBox(points)
+def makeHeighmap(path, name, size, points, heights):
+    # bail if it doesnt look write
     total_samples = len(points)
     if total_samples != len(heights):
-        print("Length don't match")
+        print("Lengths don't match")
         return
-    
-    image = Image.new("RGB", (int(size), int(size)))
-    putpixel = image.putpixel
-    imgx, imgy = image.size
-    nx = []
-    ny = []
-    nr = []
-    ng = []
-    nb = []
-
-    # Make samples data
-    for i in range(total_samples):
-        nx.append(remap(points[i][0], bbox[0], bbox[1], 0, imgx))
-        ny.append(remap(points[i][1], bbox[2], bbox[3], imgy, 0))
-
-        elev_unsigned = int(12000+heights[i])
-        nr.append(int(0))
-        ng.append(int(math.floor(elev_unsigned/255)%255))
-        nb.append(int(math.floor(elev_unsigned%255)))
 
-    # Compute Voronoi (one-to-all)
-    # TODO:
-    #   - Improve this... is very expensive 
-    for y in range(imgy):
-        for x in range(imgx):
-            dmin = math.hypot(imgx-1, imgy-1)
-            j = -1
-            for i in range(total_samples):
-                d = math.hypot(nx[i]-x, ny[i]-y)
-                if d < dmin:
-                    dmin = d
-                    j = i
-            putpixel((x, y), (nr[j], ng[j], nb[j]))
-
-    image.save(path+'/'+name+".png", "PNG")
+    # convert mercator to pixels and map pixels to height values
+    bbox = getBoundingBox(points)
+    point_heights = {}
+    for i in range(total_samples):
+        x = remap(points[i][0], bbox[0], bbox[1], 0, size))
+        y = remap(points[i][1], bbox[2], bbox[3], size, 0))
+        point_heights[(x, y)] = heights[i]
+
+    # subdivision from opencv, can do voronoi and its dual the delaunay triangulation
+    subdiv = cv2.Subdiv2D((0, 0, size, size))
+    for k, v in point_heights.iteritems():
+        subdiv.insert(v)
+    (facets, centers) = subdiv.getVoronoiFacetList([])
+
+    # an image where we will rasterize the voronoi cells
+    image = numpy.zeros((size, size, 1), dtype = 'int16')
```

Comment:
we now have a single channel texture with 16bits per pixel. where do we make the relevant changes to the shader to properly load and get the height out of the texture?